### PR TITLE
HDDS-4902 Add Layout version information to Recon datanode info API.

### PR DESCRIPTION
**Pull request by avijayanhwx**
## What changes were proposed in this pull request?
Simple change to add Layout version information to Recon datanode info API.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4902

## How was this patch tested?
Manually tested.
Added unit test.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java [hunk 1]
```diff
@@ -540,37 +540,35 @@ public void processLayoutVersionReport(DatanodeDetails datanodeDetails,
...
-            datanodeDetails.getHostName(), dnSlv, scmSlv);
-      }
+    int scmSlv = scmLayoutVersionManager.getSoftwareLayoutVersion();
+    int scmMlv = scmLayoutVersionManager.getMetadataLayoutVersion();
+    int dnSlv = layoutVersionReport.getSoftwareLayoutVersion();
+    int dnMlv = layoutVersionReport.getMetadataLayoutVersion();
+
+    // If the data node slv is > scm slv => log error condition
+    if (dnSlv > scmSlv) {
+      LOG.error("Rogue data node in the cluster : {}. " +
```
Comment (swagle) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java:550]: Rouge is generally used in the security context, I'd suggest using Outdated or something similar.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/DatanodeMetadata.java [hunk 2]
```diff
@@ -67,6 +67,9 @@
   @XmlElement(name = "buildDate")
   private String buildDate;
 
+  @XmlElement(name = "layoutVersion")
```
Comment (swagle) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/DatanodeMetadata.java:70]: General question: Why is this POJO class replicated inside recon vs having just one version in hdds-common or somewhere higher.

Comment (avijayanhwx) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/DatanodeMetadata.java:70]: You are correct, this was added as a Recon specific entity that may be formed by picking metadata pieces from multiple internal representations, and to support JAXB (I believe). We can revisit it and see if we can just reuse internal entities.

Comment (swagle) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/DatanodeMetadata.java:70]: Make sense for another time but I think having all pojos in one place makes code so much more readable.

**Review by swagle — APPROVED** [review]

**Comment by avijayanhwx** [comment]
Thanks for the review @swagle.

**Pull request merged**